lib/application.js

```diff
@@ -125,15 +125,18 @@ module.exports = class Application extends Emitter {
 
   callback() {
     const fn = compose(this.middleware);
+    const self = this;
 
     if (!this.listeners('error').length) this.on('error', this.onerror);
 
-    return (req, res) => {
+    return function handleRequest(req, res) {
       res.statusCode = 404;
-      const ctx = this.createContext(req, res);
+      const ctx = self.createContext(req, res);
       const onerror = err => ctx.onerror(err);
       onFinished(res, onerror);
-      fn(ctx).then(() => respond(ctx)).catch(onerror);
+      fn(ctx).then(function handleResponse() {
```

Comment:
@jonathanong assertion was correct. Preassigning the arrow retains the naming.

```js
const Koa = require('koa') // @2
const app = new Koa()
const myMiddleware = (ctx, next) => { ctx.body = 'foo bar' }
app.use(mw)
app.listen(2020)
```

<img width="595" alt="screen shot 2016-08-31 at 11 27 25" src="https://cloud.githubusercontent.com/assets/71882/18123452/2c20b33a-6f6e-11e6-9d80-889eee42afc3.png">

While a regular:
```js
app.use((ctx, next) => {
  ctx.body = 'foo bar'
})
```
yields:
<img width="484" alt="screen shot 2016-08-31 at 11 30 25" src="https://cloud.githubusercontent.com/assets/71882/18123520/7df59144-6f6e-11e6-8010-683f4e709436.png">